cpp/src/arrow/flight/sql/odbc/tests/connection_test.cc

```diff
@@ -101,7 +101,7 @@ TEST(ODBCHandles, TestSQLAllocAndFreeConnect) {
   ASSERT_EQ(SQL_SUCCESS, SQLFreeEnv(env));
```

Comment:
1. Please check the build failures in the CI. It seems to be complaining about `ConfigDSNW` not found. 
```
ld.lld: error: <root>: undefined symbol: ConfigDSNW
c++: error: linker command failed with exit code 1 (use -v to see invocation)
```
2. I got below 3 test failures locally, pls check:
```
[  FAILED  ] 3 tests, listed below:
[  FAILED  ] StatementTest/0.TestSQLExecDirectDataQuery, where TypeParam = arrow::flight::sql::odbc::StatementMockTest
[  FAILED  ] StatementTest/0.TestSQLExecDirectIgnoreInvalidBufLen, where TypeParam = arrow::flight::sql::odbc::StatementMockTest
[  FAILED  ] StatementTest/0.TestSQLBindColDataQuery, where TypeParam = arrow::flight::sql::odbc::StatementMockTest
```